### PR TITLE
Add ExoscalePostgreSQL composite

### PR DESCRIPTION
**Pull request by ZuGao**
* Added postgres compoiste
* Update golden tests
* Add parameter docs for the syn package

Update:
The mandatory fields are not clear yet. I set the PR reviewable as soon as I have that information.

Update 2:
Ready to be reviewed

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

### CODE REVIEW

**Comment by Kidswiss** [comment]
FWIW: with https://github.com/vshn/component-appcat/pull/33 it's possible to generate the XRD from a go struct.

But I'm fine with merging this as is and creating a tech debt to migrate later.

**Comment by ZuGao** [comment]
Let's do a full migration in another PR. I worked on this composite before #33

File: packages/composite/dbaas/exoscale/postgres.yml [hunk 1]
```diff
@@ -0,0 +1,176 @@
+classes:
+  - ...common
+
+parameters:
+
+  pkg.appcat.composite.exoscalepostgresql:
+    defaultCompositionRef: "${facts:cloud}.exoscalepostgresqls.appcat.vshn.io"
```
Comment (ccremer) [inline packages/composite/dbaas/exoscale/postgres.yml:7]: This doesn't make sense. Unlike object storage, we have a specific service that only works for exoscale. Even the kind is explicitly `ExoscalePostgreSQL` (it's not possible or sensible to use a cloudscale or AWS specific DBaaS composition for a `ExoscalePostgreSQL` for example). I see no need to have multiple compositions for this XRD, and thus this field doesn't need to be configurable.

File: docs/modules/ROOT/pages/references/composite/dbaas/exoscale/postgres.adoc [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+The parent key for all of the following parameters is`pkg.appcat.composite.dbaas.exoscale.postgres`.
+
+== `defaultCompositionRef`
+
+[horizontal]
+type:: string
+default:: `${facts:cloud}.exoscalepostgresqls.appcat.vshn.io`
+
+The default composition to use when a user creates a Claim of the composite resource.
```
Comment (ccremer) [inline docs/modules/ROOT/pages/references/composite/dbaas/exoscale/postgres.adoc:11]: As mentioned earlier, there's no need for multiple compositions, this section can be removed
```suggestion
```

File: packages/composite/dbaas/exoscale/postgres.yml [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+          connectionSecretKeys:
+            - POSTGRESQL_URL
+            - POSTGRESQL_DB
+            - POSTGRESQL_HOST
+            - POSTGRESQL_PORT
+            - POSTGRESQL_USER
+            - POSTGRESQL_PASSWORD
+            - ca.crt
+          defaultCompositionRef:
+            name: ${pkg.appcat.composite.exoscalepostgresql:defaultCompositionRef}
```
Comment (ccremer) [inline packages/composite/dbaas/exoscale/postgres.yml:29]: From #32  this should be hardcoded
```suggestion
            name: exoscale.postgres.appcat.vshn.io
```

**Review by ccremer — CHANGES_REQUESTED** [review]

File: packages/composite/dbaas/exoscale/postgres.yml (hunk 1 above)
Comment (ZuGao) [reply packages/composite/dbaas/exoscale/postgres.yml:7]: ah makes sense, we do not need for it another composition.

**Comment by ccremer** [comment]
> But I'm fine with merging this as is and creating a tech debt to migrate later.

> Let's do a full migration in another PR. I started this PR before https://github.com/vshn/component-appcat/pull/33

I disagree here. We should generate the YAML as much as possible to keep it maintainable. IMO this is important enough to do it in this PR already and not later.

**Comment by Kidswiss** [comment]
> I disagree here. We should generate the YAML as much as possible to keep it maintainable. IMO this is important enough to do it in this PR already and not later.

I've already created a tech debt that can be scheduled for the next sprint. IMHO that's the more iterative approach than to cram it into the same PR.

**Review by Kidswiss — APPROVED** [review]

**Pull request merged**